jerry-core/jcontext/jcontext.h

```diff
@@ -187,6 +190,46 @@ extern jerry_hash_table_t jerry_global_hash_table;
 
 #endif /* !CONFIG_ECMA_LCACHE_DISABLE */
 
+#else /* JERRY_ENABLE_EXTERNAL_CONTEXT */
+
+/**
+ * The following parts are for Jerry which enable external context.
+ */
+
+jerry_context_t *jerry_context_get_current_context (void);
+
+#define JERRY_CONTEXT(field) (jerry_context_get_current_context ()->field)
+
+#ifndef JERRY_SYSTEM_ALLOCATOR
+
+jmem_heap_t *jerry_context_get_current_heap (void);
+
+#define JERRY_HEAP_CONTEXT(field) (jerry_context_get_current_heap ()->field)
+
+#undef JMEM_HEAP_SIZE
+#define JMEM_HEAP_SIZE (jerry_port_get_current_instance ()->heap_size)
+
+#undef JMEM_HEAP_AREA_SIZE
+#define JMEM_HEAP_AREA_SIZE (jerry_port_get_current_instance ()->heap_size - JMEM_ALIGNMENT)
+
+#endif /* !JERRY_SYSTEM_ALLOCATOR */
+
+#ifndef CONFIG_ECMA_LCACHE_DISABLE
+
+jerry_hash_table_t *jerry_context_get_current_lcache (void);
+
+#define JERRY_HASH_TABLE_CONTEXT(field) (jerry_context_get_current_lcache ()->field)
+
+#undef ECMA_LCACHE_HASH_ROWS_COUNT
+#define ECMA_LCACHE_HASH_ROWS_COUNT (jerry_port_get_current_instance ()->lcache_row_count)
```

Comment:
Undefing ECMA_LCACHE_HASH_ROWS_COUNT after the definition above:

    ecma_lcache_hash_entry_t table[ECMA_LCACHE_HASH_ROWS_COUNT * ECMA_LCACHE_HASH_ROW_LENGTH];

This will surely not work.